src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/AbstractSpeculationAnalyzer.cs

```diff
@@ -376,10 +379,8 @@ protected bool SymbolsAreCompatible(ISymbol? symbol, ISymbol? newSymbol, bool re
                 return CompareAcrossSemanticModels(symbol, newSymbol);
             }
 
-            if (symbol.Equals(newSymbol, SymbolEqualityComparer.IncludeNullability))
```

Comment:
@RikkiGibson @cston this is the IDE change i had to make to workaround the issue.